### PR TITLE
Fix/transactions fetch

### PR DESCRIPTION
**Pull request by Lyncee59**
## Description
- Fix - Problems related to the endless scroll on each transactions pages.

## Change Type
Please enter one or more of the following: 
- Bug Fix
- Upgrades

## Testing Steps
Detail the steps required for the reviewer(s) to verify and test these changes.

## Code Checklist
- [x] Code compiles successfully (verified via `yarn start`)
- [x] No lint issues exist (verified via `yarn lint`)
- [x] New and existing unit tests pass (verified via `yarn test`)
- [x] `README.md` and other documentation is updated as needed

### CODE REVIEW

File: packages/blockchain-wallet-v4/src/redux/data/bch/selectors.js [hunk 1]
```diff
@@ -17,17 +17,15 @@ export const getTransactionHistory = path([dataPath, 'bch', 'transaction_history
...
-
 // Specific
 export const getChangeIndex = curry((xpub, state) => getAddresses(state).map(path([xpub, 'change_index'])))
 
 export const getReceiveIndex = curry((xpub, state) => getAddresses(state).map(path([xpub, 'account_index'])))
 
 export const getTotalTxPerAccount = curry((xpubOrAddress, state) => getAddresses(state).map(path([xpubOrAddress, 'n_tx'])))
 
+export const getFinalBalance = curry((address, state) => getAddresses(state).map(path([address, 'final_balance'])).map(x => x || 0))
+
```
Comment (plondon) [inline packages/blockchain-wallet-v4/src/redux/data/bch/selectors.js]: Nice

**Comment by plondon** [comment]
Noticed a problem when creating a wallet:
screen shot 2018-06-18 at 9 03 19 pm

**Comment by plondon** [comment]
BCH balance for unspendable and spendable is the same:
screen shot 2018-06-18 at 9 08 26 pm

**Comment by plondon** [comment]
BCH txs scroll fails:
screen shot 2018-06-18 at 9 12 38 pm

**Comment by plondon** [comment]
Watch only transactions no longer show up in the recent activity list.

screen shot 2018-06-18 at 9 16 44 pm

**Comment by Lyncee59** [comment]
@plondon I addressed and fixed the following bugs:
- BCH scroll issue
- Wallet creation bug (indexOf): This bug was related under the hood to a sneaky bug in the metadataBCH creation happening when we register a wallet. As now the balance is relying directly on the BCH metadata to properly filter archived accounts, we could detect this bug that was not properly populating the BCH metadata after the initial creation of BCH metadata during wallet creation.

Could not reproduce/not longer an issue:
- Balance: could not reproduce, on my side balances including unspendable match development version
- Watchonly in activityList: could not reproduce, I can see watchonly addresses and the tooltip

**Comment by plondon** [comment]
The top is this branch, the bottom is staging:

screen shot 2018-06-20 at 10 01 51 am

**Comment by plondon** [comment]
It's okay if BTC doesn't show if it's 0, think that's better actually. But BCH unspendable balance is incorrect.

**Comment by plondon** [comment]
Left is staging, right is this branch.
It's fine to not show btc/bch unspendable if it's 0, probably better, but bch unspendable is wrong:
screen shot 2018-06-20 at 10 08 30 am

File: packages/blockchain-wallet-v4-frontend/src/layouts/Wallet/MenuTop/Balance/index.js [hunk 2]
```diff
@@ -1,25 +1,17 @@
...
-import { actions } from 'data'
-import { getData } from './selectors'
+import { selectors } from 'data'
 import Template from './template'
 
 class Balance extends React.PureComponent {
   render () {
-    const { data } = this.props
-    return <Template btcUnspendableContext={data.btcUnspendableContext} bchUnspendableContext={data.bchUnspendableContext} path={data.path} />
+    return <Template path={this.props.path} />
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/layouts/Wallet/MenuTop/Balance/index.js]: Something is wrong here:
screen shot 2018-06-20 at 10 13 12 am

**Comment by plondon** [comment]
Seems really close and fixes the original issue (1031) as well! Just a few comments

File: packages/blockchain-wallet-v4/package.json [hunk 3]
```diff
@@ -79,6 +79,7 @@
     "redux": "3.6.0",
     "redux-immutable": "3.1.0",
     "redux-saga": "0.16.0",
+    "reselect": "3.0.1",
```
Comment (plondon) [inline packages/blockchain-wallet-v4/package.json]: Is this necessary in v4 core?

File: packages/blockchain-wallet-v4-frontend/src/layouts/Wallet/MenuTop/Balance/index.js (hunk 2 above)
Comment (Lyncee59) [reply packages/blockchain-wallet-v4-frontend/src/layouts/Wallet/MenuTop/Balance/index.js]: As explained earlier, there was a bug in the way we calculate the getSpendableContext and getUnspendableContext for BCH, and it was automatically excluding the legacyAddresses from the wallet payload. To double check carefully then, but these legacyAddresses were previously not included in the Spendable/Unspendable context (if not archived) and were influencing the final balance.

Comment (Lyncee59) [reply packages/blockchain-wallet-v4-frontend/src/layouts/Wallet/MenuTop/Balance/index.js]: Ok, I misunderstood the concept of Unspent balances and thought it was only about 'Imported addresses'. check my last changes

File: packages/blockchain-wallet-v4/src/redux/wallet/selectors.js [hunk 4]
```diff
@@ -27,6 +27,8 @@ export const getActiveAddresses = compose(ImtoJS, map(Address.toJS), AddressMap.
 export const getArchivedAddresses = compose(ImtoJS, map(Address.toJS), AddressMap.selectArchived, Wallet.selectAddresses, getWallet)
 export const getSpendableAddresses = compose(ImtoJS, map(Address.toJS), AddressMap.selectSpendable, Wallet.selectAddresses, getWallet)
 export const getUnspendableAddresses = compose(ImtoJS, map(Address.toJS), AddressMap.selectUnspendable, Wallet.selectAddresses, getWallet)
+export const getActiveSpendableAddresses = compose(ImtoJS, map(Address.toJS), AddressMap.selectActiveSpendable, Wallet.selectAddresses, getWallet)
+export const getActiveUnspendableAddresses = compose(ImtoJS, map(Address.toJS), AddressMap.selectActiveUnspendable, Wallet.selectAddresses, getWallet)
```
Comment (plondon) [inline packages/blockchain-wallet-v4/src/redux/wallet/selectors.js]: If an address is not active it is not spendbale

**Comment by plondon** [comment]
I also don't see why we need
```
export const getActiveSpendableAddresses = compose(ImtoJS, map(Address.toJS), AddressMap.selectActiveSpendable, Wallet.selectAddresses, getWallet)
```

if we have 

```export const getSpendableContext = compose(ImtoJS, Wallet.selectSpendableContext, getWallet)```

**Pull request merged**